### DISCUSSION TITLE
Cell cycle regress before run scanorama?

### DISCUSSION THREAD

**Discussion by Zifeng-L** [brianhie/scanorama#109] [Q&A]
Hi @brianhie , I tried to regress the cell-cycle genes effect before run scanorama. But I checked the source code of scanorama, it seemed that scanorama cannot use the result of regress out function for further analysis. How can I solve this problem?

**Comment by brianhie** [comment]
My suggestion would be to run PCA on the regressed out data before calling Scanorama, e.g., in Scanpy
```python
# Load adata, s_genes, and g2m_genes

sc.tl.score_genes_cell_cycle(adata, s_genes=s_genes, g2m_genes=g2m_genes)
sc.pp.regress_out(adata, ['S_score', 'G2M_score'])
sc.pp.scale(adata)
sc.tl.pca(adata)
sc.external.pp.scanorama_integrate(adata, 'batch', basis='X_pca')
```
Then you can use `adata.obsm['X_scanorama']` for further downstream analysis.
